shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/RouteCardHierarchy.kt

```diff
@@ -0,0 +1,184 @@
+package com.mbta.tid.mbta_app.model
+
+import com.mbta.tid.mbta_app.model.response.AlertsStreamDataResponse
+import com.mbta.tid.mbta_app.model.response.GlobalResponse
+import com.mbta.tid.mbta_app.model.response.PredictionsStreamDataResponse
+import com.mbta.tid.mbta_app.model.response.ScheduleResponse
+import io.github.dellisd.spatialk.geojson.Position
+import kotlin.time.Duration
+import kotlinx.datetime.Instant
+
+// type aliases can't be nested :(
+
+typealias ByDirection = Map<Int, RouteCardData.Leaf>
+
+private typealias ByDirectionBuilder = Map<Int, RouteCardData.LeafBuilder>
+
+private typealias ByStopIdBuilder = Map<String, RouteCardData.RouteStopDataBuilder>
+
+private typealias ByLineOrRouteBuilder = Map<String, RouteCardData.Builder>
+
+/**
+ * Contain all data for presentation in a route card. A route card is a snapshot of service for a
+ * route at a set of stops. It has the general structure: Route (orLine) => Stop(s) => Direction =>
+ * Upcoming Trips / reason for absence of upcoming trips
+ */
+data class RouteCardData(private val lineOrRoute: LineOrRoute, val stopData: List<RouteStopData>) {
```

Comment:
We have used `Transit` in some places to mean `LineOrRoute`, could adopt that here also. TBH I always call the UI component RouteCard in my head when thinking about it and just keep the caveat in my head that it also is a line card b/c GL is weird. We could also consider calling it RouteCard and just document that sometimes we are treating lines like routes sometimes.

Very to other naming ideas!

Reply:
Yeah the existing pattern made sense to me while we were doing it, but I like the idea of moving towards a spec! We should def keep that in mind as we do the barebones UX - I don't have a sense yet of whether we'd have an opportunity to make that change then while introducing new components, or if we'll just want to make a separate refactor ticket for after these data types are fully adopted

Reply:
What if we had a `RouteGroup` that contains a usually-single-element `routes: List<Route>` and a usually-null `line: Line?`

Reply:
Oh interesting, so we don't have to do as much special casing? Though I guess then we'd probably end up doing conditional expressions in the UI depending on whether line is null or not, so I'd lean towards still preferring representing them as distinct data types. But I am open to this option - maybe whoever picks up the barebones UX ticket can use that as an opportunity to re-consider if this will make things easier to work with & switch if so, since it should be a fairly minor change.